gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+
+
+class RpcAgents(object):
+
+    def __init__(self):
+        self.services = []
+        self.launchers = []
+
+    def add(self, agents):
+        for agent in agents:
+            LOG.debug(_("New RPC Agent %s" % (agent.identify())))
+        self.services.extend(agents)
+
+    def launch(self):
+        for s in self.services:
+            LOG.debug(_("Launching a rpc service %s" % (s.identify())))
+            l = os_service.launch(cfg.CONF, s)
+            self.launchers.extend([l])
+
+    def wait(self):
+        for l in self.launchers:
+            l.wait()
+
+    def report_state(self):
+        for agent in self.services:
+            agent.report_state()
+
+
+class ReportStateTask(oslo_periodic_task.PeriodicTasks):
+
+    def __init__(self, sc):
+        super(ReportStateTask, self).__init__(cfg.CONF)
+        self._sc = sc
+        pulse = loopingcall.FixedIntervalLoopingCall(
+            self.run_periodic_tasks, None, None)
+        pulse.start(interval=cfg.CONF.reportstate_interval, initial_delay=None)
+
+    @oslo_periodic_task.periodic_task(spacing=5)
+    def report_state(self, context):
+        LOG.debug(_("Report state task invoked !"))
+        self._sc.report_state()
+
+
+class PeriodicTask(oslo_periodic_task.PeriodicTasks):
+
+    def __init__(self, sc):
+        super(PeriodicTask, self).__init__(cfg.CONF)
+        self._sc = sc
+        pulse = loopingcall.FixedIntervalLoopingCall(
+            self.run_periodic_tasks, None, None)
+        pulse.start(interval=cfg.CONF.periodic_interval, initial_delay=None)
+
+    @oslo_periodic_task.periodic_task(spacing=1)
+    def periodic_sync_task(self, context):
+        LOG.debug(_("Periodic sync task invoked !"))
+        self._sc.timeout()
+
+
+class Event(object):
+
+    def __init__(self, **kwargs):
+        self.serialize = kwargs.get(
+            'serialize') if 'serialize' in kwargs else False
+        self.binding_key = kwargs.get(
+            'binding_key') if 'binding_key' in kwargs else None
+        self.id = kwargs.get('id')
+        self.key = kwargs.get('key')
+        self.data = kwargs.get('data') if 'data' in kwargs else None
+        self.handler = kwargs.get('handler') if 'handler' in kwargs else None
+        self.poll_event = None  # Not to be used by user
+        self.worker_attached = None  # Not to be used by user
+        self.last_run = None  # Not to be used by user
+        self.max_times = -1  # Not to be used by user
+
+    def identify(self):
+        return "(id=%s,key=%s)" % (self.id, self.key)
+
+
+class EventCache(object):
+
+    def __init__(self, sc):
+        self._sc = sc
+        self._cache = core_queue.Queue(sc)
+
+    def rem(self, ev):
+        self._cache.remove([ev])
+
+    def rem_multi(self, evs):
+        self._cache.remove(evs)
+
+    def add(self, ev):
+        self._cache.put(ev)
+
+    def copy(self):
+        return self._cache.copy()
+
+
+class Serializer(object):
```

Comment:
Rename to EventSerializer?
As OSM suggested, use a consistent scheme '_' separated lower case or Caps. What is the scheme used in GBP code?